modules/nw-ingress-sharding.adoc

```diff
@@ -0,0 +1,53 @@
+// Module included in the following assemblies:
+//
+// * configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-ingress-controller.adoc
+// * ingress-operator.adoc
+
+[id="nw-ingress-sharding_{context}"]
+= Configuring ingress controller sharding
+
+Ingress Controllers that have permission to view all of the labels in all
+projects can select routes to admit based on the labels. This is called Ingress
+Controller sharding. This is useful when balancing incoming traffic load among a
+set of Ingress Controllers and when isolating traffic to a specific Ingress
+Controller. For example, company A goes to one Ingress Controller and company B
+to another.
+
+.Procedure
+
+. Edit the `router-internal.yaml`:
++
+----
+# cat router-internal.yaml
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: IngressController
+  metadata:
+    name: sharded
+    namespace: openshift-ingress-operator
+  spec:
+    domain: <apps-sharded.basedomain.example.net>
+    endpointPublishingStrategy:
+      type: LoadBalancerService
+    nodePlacement:
+      nodeSelector:
+        matchLabels:
+          node-role.kubernetes.io/worker: ""
+    routeSelector:
+      matchLabels:
+        type: sharded
+  status: {}
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+----
+
+. Apply the Ingress Controller `router-internal.yaml` file:
++
+----
+# oc apply -f router-internal.yaml
+----
++
+The Ingress Operator creates a `LoadBalancerService` type and a `ClusterIPService` type.
```

Comment:
The example `resource-internal.yaml` file defines a `LoadBalancerService`-type ingress controller, for which the operator creates a `LoadBalancer`-type Service resource and a `ClusterIP`-type Service.  However, that seems like extraneous detail; more to the point, the ingress controller will select routes (in any namespace) that have the label `type: sharded`.